File: backend/NoPressure.Common/Models/Activity/NewActivity.cs
using NoPressure.Common.Enums;

namespace NoPressure.Common.Models.Activity
{
    public class NewActivity
    {
        public string Name { get; set; }
        public string? Description { get; set; }
        public string? Tag { get; set; }
        public string Color { get; set; }
        public bool IsRepeatable { get; set; }
        public ScheduleHour DirectiveTerm { get; set; }
        public int Priority { get; set; }
        public double DelayCoefficient { get; set; }
        public int Duration { get; set; }
        
        public int? TeamId { get; set; }
        public int UserId { get; set; }
    }
}

File: frontend/src/app/models/activity/new-activity.ts
import { ScheduleHour } from "../enums/ScheduleHour";

export interface NewActivity {
    userId: number;
    name: string;
    description: string;
    tag: string;
    isRepeatable: boolean;
    color: string;
    teamId: number;
    directiveTerm?: ScheduleHour;
    priority?: number;
    delayCoefficient?: number;
    duration?: number;
}